### ISSUE TITLE
Unable to install Pillow

### ISSUE THREAD

**Issue by Bodundeft** [python-pillow/Pillow#7361]
```
  31/08/2023   14:08.24   /home/mobaxterm/MyDocuments/poultrypally/Pillow   main  python3 -m pip install --upgrade pip
Requirement already satisfied: pip in /usr/local/lib/python3.9/site-packages (23.2.1)
```
This is the error that I am getting:
```
 note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for Pillow
Failed to build Pillow
ERROR: Could not build wheels for Pillow, which is required to install pyproject.toml-based projects



 
 31/08/2023   14:07.34   /home/mobaxterm/MyDocuments/poultrypally/Pillow   main  python setup.py build
running build
running build_py
creating build/lib.cygwin-3.3.5-i686-3.9
creating build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/BdfFontFile.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/BlpImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/BmpImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/BufrStubImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ContainerIO.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/CurImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/DcxImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/DdsImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/EpsImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ExifTags.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/features.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/FitsImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/FliImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/FontFile.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/FpxImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/FtexImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/GbrImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/GdImageFile.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/GifImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/GimpGradientFile.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/GimpPaletteFile.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/GribStubImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/Hdf5StubImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/IcnsImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/IcoImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/Image.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageChops.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageCms.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageColor.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageDraw.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageDraw2.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageEnhance.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageFile.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageFilter.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageFont.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageGrab.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageMath.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageMode.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageMorph.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageOps.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImagePalette.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImagePath.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageQt.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageSequence.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageShow.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageStat.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageTk.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageTransform.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImageWin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/ImtImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/IptcImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/Jpeg2KImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/JpegImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/JpegPresets.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/McIdasImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/MicImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/MpegImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/MpoImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/MspImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/PaletteFile.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/PalmImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/PcdImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/PcfFontFile.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/PcxImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/PdfImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/PdfParser.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/PixarImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/PngImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/PpmImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/PsdImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/PSDraw.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/PyAccess.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/QoiImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/SgiImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/SpiderImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/SunImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/TarIO.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/TgaImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/TiffImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/TiffTags.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/WalImageFile.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/WebPImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/WmfImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/XbmImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/XpmImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/XVThumbImagePlugin.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/_binary.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/_deprecate.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/_tkinter_finder.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/_util.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/_version.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/__init__.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
copying src/PIL/__main__.py -> build/lib.cygwin-3.3.5-i686-3.9/PIL
running egg_info
writing src/Pillow.egg-info/PKG-INFO
writing dependency_links to src/Pillow.egg-info/dependency_links.txt
writing requirements to src/Pillow.egg-info/requires.txt
writing top-level names to src/Pillow.egg-info/top_level.txt
reading manifest file 'src/Pillow.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching '*.c'
warning: no files found matching '*.h'
warning: no files found matching '*.sh'
warning: no files found matching '*.txt'
warning: no previously-included files found matching '.appveyor.yml'
warning: no previously-included files found matching '.clang-format'
warning: no previously-included files found matching '.coveragerc'
warning: no previously-included files found matching '.editorconfig'
warning: no previously-included files found matching '.readthedocs.yml'
warning: no previously-included files found matching 'codecov.yml'
warning: no previously-included files found matching 'renovate.json'
warning: no previously-included files matching '.git*' found anywhere in distribution
warning: no previously-included files matching '*.so' found anywhere in distribution
no previously-included directories found matching '.ci'
adding license file 'LICENSE'
writing manifest file 'src/Pillow.egg-info/SOURCES.txt'
running build_ext


The headers or library files could not be found for zlib,
a required dependency when compiling Pillow from source.

Please see the install instructions at:
   https://pillow.readthedocs.io/en/latest/installation.html

Traceback (most recent call last):
  File "/drives/c/Users/USER/Documents/poultrypally/Pillow/setup.py", line 988, in <module>
    setup(
  File "/usr/lib/python3.9/site-packages/setuptools/__init__.py", line 153, in setup
    return distutils.core.setup(**attrs)
  File "/usr/lib/python3.9/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.9/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.9/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/usr/lib/python3.9/distutils/command/build.py", line 135, in run
    self.run_command(cmd_name)
  File "/usr/lib/python3.9/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python3.9/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/usr/lib/python3.9/site-packages/setuptools/command/build_ext.py", line 79, in run
    _build_ext.run(self)
  File "/usr/lib/python3.9/distutils/command/build_ext.py", line 340, in run
    self.build_extensions()
  File "/drives/c/Users/USER/Documents/poultrypally/Pillow/setup.py", line 811, in build_extensions
    raise RequiredDependencyException(f)
__main__.RequiredDependencyException: zlib

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/drives/c/Users/USER/Documents/poultrypally/Pillow/setup.py", line 1005, in <module>
    raise RequiredDependencyException(msg)
__main__.RequiredDependencyException:

The headers or library files could not be found for zlib,
a required dependency when compiling Pillow from source.

Please see the install instructions at:
   https://pillow.readthedocs.io/en/latest/installation.html
```

### What are your OS, Python and Pillow versions?

* OS: 
* Python: 
* Pillow: 

```python
code goes here
```

**Comment by Bodundeft** [comment]
![WhatsApp Image 2023-08-25 at 13 26 48](https://github.com/python-pillow/Pillow/assets/101883343/28133e8d-a911-42fb-8eb6-f7f52ba66ba6)

**Comment by Bodundeft** [comment]
![WhatsApp Image 2023-08-10 at 15 07 54](https://github.com/python-pillow/Pillow/assets/101883343/bd6a2cf9-5944-42a9-8310-07fe956a3240)

![WhatsApp Image 2023-08-10 at 15 07 07](https://github.com/python-pillow/Pillow/assets/101883343/139026e2-1b5d-42f0-b645-24a995b42f19)

**Comment by Bodundeft** [comment]
What are your OS, Python and Pillow versions?
OS: Windows
Python: 4.2.2
Pillow: Name: Pillow
Version: 9.5.0
Summary: Python Imaging Library (Fork)
Home-page: https://python-pillow.org
Author: Jeffrey A. Clark (Alex)
Author-email: aclark@aclark.net
License: HPND
Location: C:\Users\USER\anaconda_3\Lib\site-packages
Requires:
Required-by: bokeh, datashader, imageio, matplotlib, scikit-image

**Comment by radarhere** [comment]
From your output, you're using Cygwin.

Pillow is complaining because one of the required dependencies, zlib, is missing. Have you tried installing it? https://www.cygwin.com/packages/summary/zlib.html

Here's a past comment from another user about what they installed if that is helpful - https://github.com/python-pillow/Pillow/issues/2860#issuecomment-628306594

> Python: 4.2.2

Just as an aside, from your output, you're using Python 3.9. I don't know what you meant by this line, since there is no Python 4.2.2

**Comment by Bodundeft** [comment]
Hello Andrew Murray,
Thanks so much

![image](https://github.com/python-pillow/Pillow/assets/101883343/7d5bd179-71fb-4421-a6bb-43129db8ec75)

![image](https://github.com/python-pillow/Pillow/assets/101883343/7dfa819f-d894-47f0-a95f-743883656036)
![image](https://github.com/python-pillow/Pillow/assets/101883343/1226a8b6-576a-476c-964c-4f2cc1167f07)
![image](https://github.com/python-pillow/Pillow/assets/101883343/e0901299-1780-4ac5-8391-cd801732b770)
![image](https://github.com/python-pillow/Pillow/assets/101883343/498a6098-5484-423b-99a0-72189e01f371)

I am still getting the errors. I have installed Cygwin on my local PC
![image](https://github.com/python-pillow/Pillow/assets/101883343/60ab8e01-265e-41a6-af16-09f1a53fa794)

**Comment by radarhere** [comment]
As that other user [suggested](https://github.com/python-pillow/Pillow/issues/2860#issuecomment-628306594), could you also install gcc-core?

**Comment by Bodundeft** [comment]
Hello Andrew Murray,
Thanks so much for your prompt response.
Please can you see the screenshots below:


![image](https://github.com/python-pillow/Pillow/assets/101883343/5e41b7ba-6a94-421c-8a34-07d8c5689efb)
![image](https://github.com/python-pillow/Pillow/assets/101883343/81b3768e-4009-4063-80a0-d7bace5e242b)
![image](https://github.com/python-pillow/Pillow/assets/101883343/b1f5ab55-1189-4291-906b-10d36961e774)
![image](https://github.com/python-pillow/Pillow/assets/101883343/3e327273-9a36-435a-9858-92303f6f9146)
![image](https://github.com/python-pillow/Pillow/assets/101883343/03be8ae2-801d-4a1d-9a35-dd5102e1da55)
![image](https://github.com/python-pillow/Pillow/assets/101883343/ee2cd525-4d64-4780-97cc-c23312323b16)
![image](https://github.com/python-pillow/Pillow/assets/101883343/575bb3a6-144c-4764-8546-767dc6cfc0d5)
![image](https://github.com/python-pillow/Pillow/assets/101883343/9e49d344-699a-4465-88b8-d81bc406b12a)
![image](https://github.com/python-pillow/Pillow/assets/101883343/2b5ab527-0e38-494f-a83d-f9c3ad17d182)
![image](https://github.com/python-pillow/Pillow/assets/101883343/f13673a9-ef1b-40f4-9b82-8cf6d0b7d043)
![image](https://github.com/python-pillow/Pillow/assets/101883343/3a6c1a27-ddad-45da-a8fa-4baa2a55d91f)
![image](https://github.com/python-pillow/Pillow/assets/101883343/2e830101-3311-4ad8-8b23-f350c8078c8a)
![image](https://github.com/python-pillow/Pillow/assets/101883343/8f49cb97-e725-4276-b210-4ebd20c18557)
![image](https://github.com/python-pillow/Pillow/assets/101883343/729fded1-3ebb-4755-b8cf-92efe5ec0e84)


I have installed gcc-core complier but I am still get the error.

**Comment by radarhere** [comment]
To be clear, the error is changing each time, so this is progressing.

I think installing python39-devel would fix this screen
<img src="https://user-images.githubusercontent.com/101883343/264962448-2e830101-3311-4ad8-8b23-f350c8078c8a.png" width="300" />

And I think installing libjpeg-devel would fix this screen
<img src="https://user-images.githubusercontent.com/101883343/264962671-729fded1-3ebb-4755-b8cf-92efe5ec0e84.png" width="300" />

**Comment by Bodundeft** [comment]
Hello Andrew Murray,
Thanks so much for your prompt response.
This  python39-devel fixed it and it said that I have already installed  libjpeg-devel
![image](https://github.com/python-pillow/Pillow/assets/101883343/608e7391-072e-4f0f-b0b2-cadbd3e0cc5f)

Thanks so much, you rock
You are the best. One more thing, I am new to this Django, I hope that I can be reaching out to you?

**Comment by radarhere** [comment]
I've not used Django. This is a repository for Pillow and Pillow-related questions.

If you would to know more about Django, you can go to https://github.com/django/django, [ask questions on StackOverflow](https://stackoverflow.com/questions/tagged/django), or https://github.com/django/django#readme
> - Join the #django channel on irc.libera.chat. Lots of helpful people hang out there. [Webchat is available](https://web.libera.chat/#django).
> - Join the django-users mailing list, or read the archives, at https://groups.google.com/group/django-users.
> - Join the [Django Discord community](https://discord.gg/xcRH6mN4fa).
> - Join the community on the [Django Forum](https://forum.djangoproject.com/).

**Comment by Bodundeft** [comment]
I tried to Perform Django Data Migration using commands below:

python manage.py makemigrations
python manage.py migrate

![image](https://github.com/python-pillow/Pillow/assets/101883343/d0898c39-1a1c-4c41-bb6e-c8a2399f01a1)

![image](https://github.com/python-pillow/Pillow/assets/101883343/98ae61f1-1a96-409f-b238-b54725807cc2)

**Comment by Bodundeft** [comment]
Thanks so much for this.

**Comment by radarhere** [comment]
The error seems to fairly straightforwardly say that SECRET_KEY is missing as an environment variable. Googling, I found https://stackoverflow.com/questions/59823749/undefinedvalueerror-secret-key-deploying-docker-to-digitalocean, which says that maybe the answer to your question is that you're missing it from .env or .settings.ini

But I really have no idea about Django.

Closing, as these are no further Pillow questions.
